### PR TITLE
Add support for adding/removing development dependencies

### PR DESCRIPTION
**Pull request by ibraheemdev**
## Summary

Support adding/removing dependencies from `tool.uv.dev-dependencies` with `uv add/remove --dev`.

Part of https://github.com/astral-sh/uv/issues/3959.

### CODE REVIEW

File: crates/uv/src/cli.rs [hunk 1]
```diff
@@ -1634,6 +1638,10 @@ pub(crate) struct RemoveArgs {
     #[arg(required = true)]
     pub(crate) requirements: Vec<PackageName>,
 
+    /// Remove the requirements from development dependencies.
+    #[arg(long)]
+    pub(crate) dev: bool,
+
```
Comment (zanieb) [inline crates/uv/src/cli.rs:1644]: If I try to remove a dependency that's not present in the production dependencies but in the development dependencies should we just infer that you meant dev? Inference might be too far, but we should at least provide a hint suggesting `--dev` is what you wanted.

Comment (ibraheemdev) [reply crates/uv/src/cli.rs:1644]: See https://github.com/astral-sh/uv/pull/4327/files#diff-ba8cc7de2d3484ebe70b612b0ecb1ec65d09e426593158e93382cfc0af426640R47.

Comment (zanieb) [reply crates/uv/src/cli.rs:1644]: Wow you're on it :)

Comment (zanieb) [reply crates/uv/src/cli.rs:1644]: I guess I should actually read the pull request before I comment on the UX haha

File: crates/uv/tests/edit.rs [hunk 2]
```diff
@@ -167,7 +167,7 @@ fn add_git() -> Result<()> {
      + sniffio==1.3.1
     "###);
 
-    uv_snapshot!(context.filters(), context.add(&["uv-public-pypackage @ git+https://github.com/astral-test/uv-public-pypackage@0.0.1"]), @r###"
+    uv_snapshot!(context.filters(), context.add(&["uv-public-pypackage @ git+https://github.com/astral-test/uv-public-pypackage@0.0.1"], false), @r###"
```
Comment (zanieb) [inline crates/uv/tests/edit.rs:170]: Why this change?

File: crates/uv/tests/edit.rs [hunk 3]
```diff
@@ -374,6 +374,128 @@ fn add_unnamed() -> Result<()> {
...
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(indoc! {r#"
+        [project]
+        name = "project"
+        version = "0.1.0"
+        requires-python = ">=3.12"
+        dependencies = []
+    "#})?;
+
+    uv_snapshot!(context.filters(), context.add(&["anyio==3.7.0"], true), @r###"
```
Comment (zanieb) [inline crates/uv/tests/edit.rs:391]: Similar to above, why are we toggling the Windows filters?

Comment (zanieb) [reply crates/uv/tests/edit.rs:391]: I'm a little confused, where's the `--dev` flag?

File: crates/uv/tests/edit.rs [hunk 4]
```diff
@@ -374,6 +374,128 @@ fn add_unnamed() -> Result<()> {
...
+
+        [tool.uv]
+        dev-dependencies = [
+            "anyio==3.7.0",
+        ]
+        "###
+        );
+    });
+
+    let lock = fs_err::read_to_string(context.temp_dir.join("uv.lock"))?;
```
Comment (zanieb) [inline crates/uv/tests/edit.rs:428]: Why are we testing lock and sync during `add --dev`?

File: crates/uv/tests/edit.rs (hunk 3 above)
Comment (zanieb) [reply crates/uv/tests/edit.rs:391]: Ah okay I see now that you're using `true`/`false` to toggle dev not the `windows_filters`.

File: crates/uv/tests/common/mod.rs [hunk 5]
```diff
@@ -321,7 +321,7 @@ impl TestContext {
     }
 
     /// Create a `uv add` command for the given requirements.
-    pub fn add(&self, reqs: &[&str]) -> std::process::Command {
+    pub fn add(&self, reqs: &[&str], dev: bool) -> std::process::Command {
```
Comment (zanieb) [inline crates/uv/tests/common/mod.rs:324]: This isn't our usual pattern, I'd specify `.arg("--dev")` on each invocation instead.

File: crates/uv/tests/edit.rs (hunk 3 above)
Comment (zanieb) [reply crates/uv/tests/edit.rs:391]: See https://github.com/astral-sh/uv/pull/4327#discussion_r1640130152

**Review by zanieb — APPROVED** [review]
Implementation looks good, some questions about the tests.

File: crates/uv/tests/common/mod.rs (hunk 5 above)
Comment (ibraheemdev) [reply crates/uv/tests/common/mod.rs]: That's nicer, updated.

File: crates/uv/tests/edit.rs (hunk 4 above)
Comment (zanieb) [reply crates/uv/tests/edit.rs:428]: Ah okay that seems reasonable. Thanks! A comment may be nice, e.g. "Adding implies a lock and sync including development dependencies"
